core/src/testUtils/groovy/nu/yona/server/YonaServer.groovy

```diff
@@ -172,4 +182,64 @@ class YonaServer
 	{
 		DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSZ").format(dateTime)
 	}
+
+	static def relativeDateTimeStringToZonedDateTime(relativeDateTimeString)
+	{
+		def fields = relativeDateTimeString.tokenize(' ')
+		assert fields.size() <= 3
+		assert fields.size() > 0
+		int parsedFields = 0
+		int weekOffset = 0
+		int dayOffset = 0
+
+		LocalDateTime now = LocalDateTime.now()
+		switch (fields.size())
+		{
+			case 3:
+				assert fields[0].startsWith("W")
+				weekOffset = Integer.parseInt(fields[0].substring(1))
+				parsedFields++
+			// Fall through
+			case 2:
+				int weekDay = DateTimeFormatter.ofPattern("eee").parse(fields[parsedFields]).get(ChronoField.DAY_OF_WEEK)
```

Comment:
Done, http://jira.yona.nu/browse/YD-259

On Sat, Jun 4, 2016 at 3:29 PM, Bert Roos <notifications@github.com> wrote:

> In core/src/testUtils/groovy/nu/yona/server/YonaServer.groovy
> <https://github.com/yonadev/yona-server/pull/183#discussion_r65802556>:
>
> > +		assert fields.size() <= 3
> > +		assert fields.size() > 0
> > +		int parsedFields = 0
> > +		int weekOffset = 0
> > +		int dayOffset = 0
> > +
> > +		LocalDateTime now = LocalDateTime.now()
> > +		switch (fields.size())
> > +		{
> > +			case 3:
> > +				assert fields[0].startsWith("W")
> > +				weekOffset = Integer.parseInt(fields[0].substring(1))
> > +				parsedFields++
> > +			// Fall through
> > +			case 2:
> > +				int weekDay = DateTimeFormatter.ofPattern("eee").parse(fields[parsedFields]).get(ChronoField.DAY_OF_WEEK)
>
> I have tested quite a few of them last week, but honestly, I don't know
> this one.
>
> If you're not convinced, please create a ticket to verify it later.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/yonadev/yona-server/pull/183/files/5bda2cadfe8cd9af0dc5c71b05d57f4b8a221f82#r65802556>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACrzJot4payfG5Psumkg5G60GuUsz46Sks5qIX2ygaJpZM4Im7xS>
> .
>